.github/workflows/test-and-lint-js.yml

```diff
@@ -0,0 +1,43 @@
+name: Test & lint JS codebase
+
+on:
+  pull_request:
+    branches: [ "main" ]
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths:
+      - '**.js'
+      - '**.ts'
+      - '**.vue'
+      - 'package.json'
+      - 'package.lock'
+
+jobs:
+  test-and-lint-js:
+    name: Test & lint JS codebase
+    timeout-minutes: 10
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-24.04
+    env:
+      # renovate: datasource=node-version depName=node
+      NODE_VERSION: 24.4.0
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2 # https://github.com/actions/checkout
+
+      - name: Cache dependencies
+        uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3 # https://github.com/actions/cache
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-npm-dependencies-${{ hashFiles('package.lock') }}
+          restore-keys: ${{ runner.os }}-npm-dependencies
+
+      - name: Set up node
+        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0 # https://github.com/actions/setup-node
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Instal npm dependencies
+        run: npm clean-install
+
+      - name: Run JS linter
+        run: npm run lint
```

Comment:
Please take care about adding new line at the end of the file. Please check other files 👀